### ISSUE TITLE
Add support for Network Device Naming in Hyper-V

### ISSUE THREAD

**Issue by janegilring** [VirtualEngine/Lability#125]
In Windows 10 and Windows Server 2016, Microsoft have introduced a set of new capabilities around managing virtual network adapters, more info [here](https://blogs.msdn.microsoft.com/virtual_pc_guy/2016/06/10/network-device-naming-in-hyper-v/).

This feature can be leveraged to mitigate the network adapter naming issue which often occurs when working with DSC configurations where network adapter names is hard coded.

For example, in the bootstrap file, it could be added a few lines to check whether the property "Hyper-V Network Adapter Name" has a value, and if so rename the network adapter inside the guest operating system to the same name.

Another possibilty could be creating a separate DSC Resource for this.

I'm open to contribute, but I'd like some views on the best way to do this.

We must also remember to enable Device Naming when creating VM network adapters in the Lability module.

**Comment by iainbrighton** [comment]
@janegilring Think out loud here. I think we can look at incorporating the Hyper-V Device Naming functionality when we drop support for v4 and target Server 2016/Windows 10.

Realistically though, I think the effort should be put into a DSC resource that would benefit everyone. @PlagueHO is proposing an [AdapterName](https://github.com/PowerShell/xNetworking/issues/115) resource and I also think that @TravisEz13 had some prototype code. Somewhere..?

**Comment by janegilring** [comment]
I totally agree on that thought, having a DSC Resource for this independently of Lability would benefit other scenarios. And Lability could still use it without much (or any?) change.
